OpenRA.Mods.Common/Scripting/Media.cs

```diff
@@ -22,7 +22,7 @@ public static class Media
 		public static void PlayFMVFullscreen(World w, string movie, Action onComplete)
 		{
 			var playerRoot = Game.OpenWindow(w, "FMVPLAYER");
-			var player = playerRoot.Get<VqaPlayerWidget>("PLAYER");
+			var player = playerRoot.Get<VideoPlayerWidget>("PLAYER");
```

Comment:
I don't understand the reason for this refactor. You create a new "generic" `VideoPlayerWidget`, but the only format supported is .vqa anyway. It may be simpler to move everything touching video support to Mods.Cnc to avoid common depending on cnc, which seems to be your initial motivation.

Reply:
This is running ahead a bit and is setup to suit with, say, an MPEG-2 decoder at hand, whose mp2 support can stay within Mods.Common.

Reply:
Yeah, the idea is to avoid having to move back Media and MediaGlobal once we support some generic, non-proprietary video format.

Reply:
What I mean is that you seem to be handling inheritance the wrong way. You are referencing the base/placeholder class here. It doesn't automatically resolve as the correct `VqaPlayerWidget` automagically. You only get the generic stub, so I am not surprised it doesn't work as intended.

My suggestion would be to have a fully functional `VideoPlayerWidget` and let it decide for the file format by looking for interfaces that check for magic bytes or file endings and then get back a `VqaReader` stream. Or if you want to keep `VqaPlayerWidget` then let the media routine here pick the correct widget based on what file it is.

Reply:
See how https://github.com/OpenRA/OpenRA/blob/bleed/OpenRA.Game/Sound/Sound.cs handles first the check if it's the turn of the audio loader and then returns a stream.